### PR TITLE
Fixed Sonar issues in EdifactService implementation

### PR DESCRIPTION
**Pull request by vbergmann**

### CODE REVIEW

File: src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java [hunk 1]
```diff
@@ -38,4 +38,19 @@ public EdiDiffDetailType getEdiDiffType() {
...
+    @Override
+    public int hashCode() {
+        return 31 * super.hashCode() + ((ediType == null) ? 0 : ediType.hashCode());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (!super.equals(obj) || getClass() != obj.getClass())
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java]: `super.equals` should never be true, as equality should always only match on SAME class (not instanceof). Please replace with `obj == null` check. Also, please add the curly braces here.

File: src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java [hunk 2]
```diff
@@ -38,4 +38,19 @@ public EdiDiffDetailType getEdiDiffType() {
...
+            return false;
+        EdiDiffDetailImpl that = (EdiDiffDetailImpl) obj;
+        return (super.equals(that) && ediType == that.ediType);
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java]: See above - do not call super.equals.

File: src/main/java/org/aludratest/content/edifact/edifatto/EdifattoContent.java [hunk 3]
```diff
@@ -48,6 +48,7 @@
 
     /** Constructor. */
     public EdifattoContent() {
+        // Empty constructor: No members to initialize
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdifattoContent.java]: Why having this constructor at all?

File: src/main/java/org/aludratest/content/edifact/edifatto/EdifattoUtil.java [hunk 4]
```diff
@@ -25,6 +25,10 @@
  * @author Volker Bergmann */
 public class EdifattoUtil {
 
+    private EdifattoUtil() {
+        // private constructor to prevent instantiation of this utility class
+    }
+
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdifattoUtil.java]: I think SonarQube also wants the class to be "final" in this case.

File: src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java (hunk 1 above)
Comment (vbergmann) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java]: Done

File: src/main/java/org/aludratest/content/edifact/edifatto/EdifattoContent.java (hunk 3 above)
Comment (vbergmann) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdifattoContent.java]: done

File: src/main/java/org/aludratest/content/edifact/edifatto/EdifattoUtil.java (hunk 4 above)
Comment (vbergmann) [inline src/main/java/org/aludratest/content/edifact/edifatto/EdifattoUtil.java]: done

**Pull request merged**